Sources/DoctorPretty/RenderPretty.swift

```diff
@@ -100,19 +100,35 @@ extension Doc {
     static func nicest1(nesting: IndentLevel, column: ColumnCount, pageWidth: Width, ribbonWidth: RibbonWidth) -> (SimpleDoc, () -> SimpleDoc) -> SimpleDoc {
         return { d1, d2 in
             let wid = min(pageWidth - column, ribbonWidth - column + nesting)
-            
+
             func fits(prefix: Int, w: Int, doc: SimpleDoc) -> Bool {
-                switch (prefix, w, doc) {
-                case let (_, w, _) where w < 0: return false
-                case (_, _, .empty): return true
-                case let (m, w, .char(_, x)):
-                    return fits(prefix: m, w: w-1, doc: x)
-                case let (m, w, .text(length, _, x)):
-                    return fits(prefix: m, w: w-length, doc: x)
-                case (_, _, .line(_, _)): return true
+                var _w = w
+                var _doc = doc
+
+                while true {
```

Comment:
here's where I unrolled the recursion to just a loop. it ain't pretty, but it works. I was also able to simplify some of the logic inside.